src/components/Hooks/useLocalStorage.js

```diff
@@ -0,0 +1,39 @@
+    // Step 1. import useState from react. I didn't import React from react because we arent using jsx so theres no need to import React. 
+    import { useState } from 'react';
+
+    // Step 2. Build the function 'useLocalStorage' 
+    // This works very similarly to the useState hook but when I save my values I also want to put them in local storage so that if I do a refresh it checks and sees if I already have a value saved there and be able to use my preference.
+    // I also need a 'key' whenever i'm storing things into local storage. So i need to pass that through the array as well. 
+export const useLocalStorage = (key, initialValue) => {
+    // with useState I can use an initial value and get back and array with the value and then a function to change that value
+    // useState is a function and it takes an initial value and it returns an array with a value and then a function to set the value. This is what I want useLocalStorage to be.
+    // i.e. useState(initialValue) -----> [value, setValue]
+    
+    // Since I want to use my stored value as my useState and not my initialValue. useState has a unique feature that instead of using an initialValue I can give useState a callback and what that callback returns is going to be used as the initial value for my useState.
+    // I can give this a callback and then I can return whatever value I want useState to use as the initial value for storedValue
```

Comment:
by passing a CB the computed value is what is returned every time instead of just the initial value